test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs

```diff
@@ -1507,6 +1507,63 @@ public async Task GivenImportWithCancel_ThenTaskShouldBeCanceled()
             Assert.Equal(HttpStatusCode.BadRequest, fhirException.StatusCode);
         }
 
+        [Fact]
+        public async Task GivenImportTriggeredWithMultipleFilesWithCancel_ThenTaskShouldBeCanceled()
+        {
+            _metricHandler?.ResetCount();
+
+            string patientNdJsonResource = Samples.GetNdJson("Import-SinglePatientTemplate");
+            var patients = new List<InputResource>();
+
+            for (int i = 0; i <= 100; i++)
+            {
+                string resourceId = Guid.NewGuid().ToString("N");
+                string patientNdJsonResourceN = patientNdJsonResource.Replace("##PatientID##", resourceId);
+
+                (Uri location, string _) = await ImportTestHelper.UploadFileAsync(patientNdJsonResourceN, _fixture.StorageAccount);
+
+                var inputResource = new InputResource()
+                {
+                    Url = location,
+                    Type = "Patient",
+                };
+
+                patients.Add(inputResource);
+            }
+
+            var request = new ImportRequest()
+            {
+                InputFormat = "application/fhir+ndjson",
+                InputSource = new Uri("https://other-server.example.org"),
+                StorageDetail = new ImportRequestStorageDetail() { Type = "azure-blob" },
+                Input = patients,
+                Mode = ImportMode.InitialLoad.ToString(),
+            };
+
+            Uri checkLocation = await ImportTestHelper.CreateImportTaskAsync(_client, request);
+
+            await Task.Delay(TimeSpan.FromMinutes(1));
```

Comment:
We could reduce test wait time by polling the Content-Location and wait for a body to be returned.